nuxeo-features/nuxeo-platform-audit/nuxeo-platform-audit-mongodb/src/main/java/org/nuxeo/mongodb/audit/MongoDBAuditBackend.java

```diff
@@ -0,0 +1,356 @@
+import org.nuxeo.ecm.platform.audit.api.FilterMapEntry;
+import org.nuxeo.ecm.platform.audit.api.LogEntry;
+import org.nuxeo.ecm.platform.audit.api.query.AuditQueryException;
+import org.nuxeo.ecm.platform.audit.api.query.DateRangeParser;
+import org.nuxeo.ecm.platform.audit.service.AbstractAuditBackend;
+import org.nuxeo.ecm.platform.audit.service.AuditBackend;
+import org.nuxeo.ecm.platform.audit.service.BaseLogEntryProvider;
+import org.nuxeo.ecm.platform.audit.service.NXAuditEventsService;
+import org.nuxeo.ecm.platform.audit.service.extension.AuditBackendDescriptor;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.model.DefaultComponent;
+import org.nuxeo.runtime.mongodb.MongoDBComponent;
+import org.nuxeo.runtime.mongodb.MongoDBConnectionService;
+import org.nuxeo.runtime.mongodb.MongoDBSerializationHelper;
+import org.nuxeo.runtime.services.config.ConfigurationService;
+
+import com.mongodb.client.FindIterable;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.Sorts;
+
+/**
+ * Implementation of the {@link AuditBackend} interface using MongoDB persistence.
+ *
+ * @since 9.1
+ */
+public class MongoDBAuditBackend extends AbstractAuditBackend implements AuditBackend {
+
+    private static final Log log = LogFactory.getLog(MongoDBAuditBackend.class);
+
+    public static final String AUDIT_DATABASE_ID = "audit";
+
+    public static final String COLLECTION_NAME_PROPERTY = "nuxeo.mongodb.audit.collection.name";
+
+    public static final String DEFAULT_COLLECTION_NAME = "audit";
+
+    public static final String SEQ_NAME = "audit";
+
+    protected MongoCollection<Document> collection;
+
+    protected MongoDBLogEntryProvider provider = new MongoDBLogEntryProvider();
+
+    public MongoDBAuditBackend(NXAuditEventsService component, AuditBackendDescriptor config) {
+        super(component, config);
+    }
+
+    @Override
+    public int getApplicationStartedOrder() {
+        DefaultComponent component = (DefaultComponent) Framework.getRuntime().getComponent(MongoDBComponent.NAME);
+        return component.getApplicationStartedOrder() + 1;
+    }
+
+    @Override
+    public void onApplicationStarted() {
+        log.info("Activate MongoDB backend for Audit");
+        // First retrieve the collection name
+        ConfigurationService configurationService = Framework.getService(ConfigurationService.class);
+        String collName = configurationService.getProperty(COLLECTION_NAME_PROPERTY, DEFAULT_COLLECTION_NAME);
+        // Get a connection to MongoDB
+        MongoDBConnectionService mongoService = Framework.getService(MongoDBConnectionService.class);
+        MongoDatabase database = mongoService.getDatabase(AUDIT_DATABASE_ID);
+        collection = database.getCollection(collName);
+        // TODO migration ?
+    }
+
+    @Override
+    public void onShutdown() {
```

Comment:
`onShutDown` is deprecated in favour of `onApplicationStopped`, I planned to replace it with NXP-21860, but could be interesting to have the proper implementation as this class is new on platform.